src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/IndirectionNode.cs

```diff
@@ -12,19 +12,40 @@ namespace ILCompiler.DependencyAnalysis
     /// </summary>
     public class IndirectionNode : ObjectNode, ISymbolNode
     {
-        private ISymbolNode _indirectedNode;
+        public class IndirectionNodeData
```

Comment:
This could just be a private helper struct in `NodeFactory` and have the constructor accept ISymbolNode and an int offset. No need for a public nested type.

The argument for that being that the *need* for a data structure that groups this is at the `NodeFactory`. The node itself doesn't care how many arguments the ctor has (this is similar to the `MethodKey` struct we already have in `NodeFactory`).